dex/src/state.rs

```diff
@@ -2686,8 +2691,201 @@ impl State {
-        unimplemented!()
+    fn process_send_take(args: account_parser::SendTakeArgs) -> DexResult {
+
+        let account_parser::SendTakeArgs {
+            instruction,
+            signer,
+            mut event_q,
+            mut req_q,
+            mut order_book_state,
+            coin_wallet,
+            pc_wallet,
+            coin_vault,
+            pc_vault,
+            spl_token_program,
+            fee_tier,
+            vault_signer
+        } = args;
+
+        // Question about architecture of max coin, max price, min, etc
+        let limit_price = instruction.limit_price;
+        let mut limit = instruction.limit;
+        let min_coin_qty = instruction.min_coin_qty;
+        let min_native_pc_qty = instruction.min_native_pc_qty;
+
+        let order_type = OrderType::ImmediateOrCancel;
+        let self_trade_behavior = SelfTradeBehavior::AbortTransaction;
+        let native_pc_qty_locked;
+
+        let order_id = req_q.gen_order_id(instruction.limit_price.get(), instruction.side); 
+        let mut proceeds = RequestProceeds::zero();
+        
+        match instruction.side {
+            Side::Bid => {
+                let lock_qty_native = instruction.max_native_pc_qty_including_fees;
+                native_pc_qty_locked = Some(lock_qty_native);
+            }
+            Side::Ask => {
+                native_pc_qty_locked = None;
+            }
+        };
+
+        let request = RequestView::NewOrder {
+            side: instruction.side,
+            order_type,
+            order_id,
+            fee_tier,
+            self_trade_behavior,
+            // Pass dummy account - system program pubkey
+            owner: solana_program::system_program::ID.to_aligned_bytes(),
+            // Pass 0 
+            owner_slot: 0,
+            max_coin_qty: instruction.max_coin_qty,
+            native_pc_qty_locked,
+            client_order_id: None,
+        };
+
+        let _unfilled_portion = order_book_state.process_orderbook_request(
+            &request,
+            &mut event_q,
+            &mut proceeds,
+            &mut limit,
+        )?;
+
+        let pc_lot_size = order_book_state.market_state.pc_lot_size;
+        let coin_lot_size = order_book_state.market_state.coin_lot_size;
+
+        let RequestProceeds {
+            coin_unlocked,
+            coin_credit, // 0
+
+            native_pc_unlocked, // 0
+            native_pc_credit,
+
+            coin_debit,
+            native_pc_debit, // 0
+        } = proceeds;
+
+        let mut abort= false;
+
+        match instruction.side {
+            Side::Bid => {
+                if coin_credit < min_coin_qty {
+                    abort = true;
+                };
+            },
+            Side::Ask => {
+                if native_pc_credit < min_native_pc_qty {
+                    abort = true;
+                }
+            }
+        }
+
+        if abort {
+            return Ok(());
```

Comment:
I feel like this should error. If you return `Ok` the the state is going to get corrupted without rolling back